File: tests/conformance/sandbox-tools/fetch_wildcard.py
tool = {
    "description": "Tool with wildcard network permission",
    "input": {},
    "permissions": {
        "network": ["*.localhost"],
    },
}


async def execute(args, ctx):
    try:
        res = await ctx.fetch("http://localhost:18923/test")
        return {"status": "ok", "domain": "localhost"}
    except Exception as e:
        return {"status": "error", "message": str(e)}

File: tests/conformance/sandbox-tools/fetch_wildcard.js
export default {
  description: "Tool with wildcard network permission",
  input: {},
  permissions: {
    network: ['*.localhost'],
  },
  execute: async (args, ctx) => {
    try {
      const res = await ctx.fetch('http://localhost:18923/test');
      const body = await res.text();
      return { status: 'ok', domain: 'localhost' };
    } catch (err) {
      return { status: 'error', message: err.message };
    }
  },
};
